pulpcore/pulpcore/tasking/services/worker_watcher.py

```diff
@@ -76,17 +77,20 @@ def check_celery_processes():
     now = timezone.now()
     oldest_heartbeat_time = now - timedelta(seconds=TASKING_CONSTANTS.PULP_PROCESS_TIMEOUT_INTERVAL)
 
-    for worker in Worker.objects.filter(last_heartbeat__lt=oldest_heartbeat_time, online=True):
+    for worker in Worker.objects.filter(last_heartbeat__lt=oldest_heartbeat_time,
+                                        gracefully_stopped=False):
         msg = _("Worker '%s' has gone missing, removing from list of workers") % worker.name
         _logger.error(msg)
 
         mark_worker_offline(worker.name)
 
     worker_count = Worker.objects.exclude(
-        name__startswith=TASKING_CONSTANTS.RESOURCE_MANAGER_WORKER_NAME).filter(online=True).count()
+        name__startswith=TASKING_CONSTANTS.RESOURCE_MANAGER_WORKER_NAME)\
```

Comment:
Similar for this, can this be refactored to use `Worker.objects.online_workers()`. I think the current approach would include non-gracefully stopped workers incorrectly without a change.

Reply:
Nope, because this code only wants "workers" and doesn't care about resource managers, and ```Worker.objects.online_workers()``` returns both.

I think it would be entirely reasonable to rewrite ```Worker.objects.online_workers()``` to exclude the resource manager but then the status API couldn't use it... so it would just be shifting the problem somewhere else.

Alternatively, we could have a kwarg to toggle whether or not to exclude the resource manager from the query.

Reply:
Querysets are chainable so you can Worker.objects.online_workers().exclude(name__startswith=TASKING_CONSTANTS.RESOURCE_MANAGER_WORKER_NAME)` without having to "build" those capabilities into the `online_workers` manager method itself.